LICENSE-3rdparty.csv

```diff
@@ -27,7 +27,11 @@ require,tar,ISC,Copyright Isaac Z. Schlueter and Contributors
 require,url-parse,MIT,Copyright 2015 Unshift.io Arnout Kazemier the Contributors
 require,whatwg-fetch,MIT,Copyright 2014-2016 GitHub Inc.
 dev,@babel/core,MIT,Copyright 2014-present Sebastian McKenzie and other contributors
+dev,@babel/plugin-proposal-decorators,MIT,Copyright 2014-present Sebastian McKenzie and other contributors
```

Comment:
Why is additional tooling needed for this? The specification can be implemented and tested in pure JS.

Reply:
You can _implement_ it in pure JS, and the entirety of the runtime code I added doesn't use the `@` syntax.

However I don't think it can be _tested_ without the babel plugin:
- If you want to use the `@` syntax in your tests, you need the babel plugin.
- If you're OK just calling the function directly in tests, the babel plugin could be removed.

I personally feel like using the `@` syntax in tests is important, both because that is how it will be consumed by library users and because there would be no other way to ensure the source code conforms to what babel/typescript are expecting.

Reply:
Actually, after reading [this](https://github.com/tc39/proposal-decorators#how-should-this-new-proposal-be-used-in-transpilers-when-its-implemented) it sounds like there are other issues at play here:

* The spec cannot be implemented in pure JavaScript until the feature actually lands in the language.
* There is no guarantee that Babel will not change its implementation since the spec is no longer in JavaScript.
* Unrelated, but there is also the consideration that this project was never optimized to be transpiled efficiently for testing.

Said that, I'm not really sure what would be the best approach to ensure that this works not only now but in the future as well. According to the spec, it sounds like the best approach is to ship decorators as another module. This would make the feature more complicated to implement, but definitely more feature proof and spec compliant.

Or it could be shipped in the same package, but as a submodule like `dd-trace/decorators`. This would mean however that my original plan of using `tracer.wrap` may not work. There is also the concern of `wrap` being a built-in decorator so to avoid collisions it may be better to stick to `trace`.

Either way, I'm also not sure how tests should be setup for this so that the rest of the test suite doesn't need to be transpiled.

Thoughts?